app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -116,6 +116,11 @@ def update_auth_header
   private
 
   def refresh_headers
+    # At this point restore the attributes so we can obtain the lock
+    # otherwise an exception is thrown while trying to obtain the lock
+    # with unsaved data on the resource.
+    @resource.restore_attributes unless const_defined?('Mongoid')      
```

Comment:
I think you have to use `defined?` instead, `const_defined?` is with `Module.const_defined?("something")`